feat(textarea): novo componente

nicoleoliveira opened the pull request:
### Textarea - novo componente

Novo componente do tipo textarea.
Versão básica e unitária do componente.

**Revisão, testar no:**

- labs do storybook
- live demo

PR animalia brand: animaliads/animalia-brand#9

nicoleoliveira committed:
feat(textarea): estrutura inicial

nicoleoliveira committed:
feat(textarea): estrutura inicial

nicoleoliveira committed:
feat: estilo

nicoleoliveira committed:
feat(textarea): implementação das propriedades

nicoleoliveira committed:
feat(textarea): listeners

nicoleoliveira committed:
Merge branch 'next' into textarea

nicoleoliveira committed:
feat(textarea): correção dos listeners e value

nicoleoliveira committed:
feat(textarea): listener focus

nicoleoliveira committed:
feat(textarea): metodos

nicoleoliveira committed:
feat(textarea): correção da mudança dinamica das propriedades

nicoleoliveira committed:
feat(textarea): documentação

nicoleoliveira committed:
feat: atualização da borda

nicoleoliveira committed:
feat: ajuste no estilo

nicoleoliveira committed:
Update pr_ci.yml

nicoleoliveira committed:
Update pr_ci.yml

edioli commented on packages/textarea/src/stories/general.mdx:
```diff
@@ -0,0 +1,3 @@
+# Radio
```
~Radio~ Text Area

edioli reviewed (changes requested):
- No estado **disabled**, por alguma razão não está pegando a cor do `--background-disabled`. Parece que a cor da variação do read-only está sobrepondo a do disabled.

- Ficou faltando a opção de incluir um placeholder

- No Safari, não está deixando redimensionar horizontalmente, como é original do componente (No Chrome está funcionando)

![text-area-safari](https://user-images.githubusercontent.com/75502935/131923396-dd88ad8e-353b-4b1b-9b95-c3251bb0a56a.gif)

edioli commented on packages/textarea/src/stories/custom-properties.mdx:
```diff
@@ -0,0 +1,21 @@
+
+O componente de textarea possui propriedades customizáveis, que possibilitam fazer alterações, de acordo com a marca ou o tema a ser utilizado. Com isso, garantimos uma consistência ao Design System, sem perder a identidade de cada marca.
+
+Possui cinco estados, que se modificam de acordo com a interação do usuário, sendo eles: _Normal_, _Hover_, _Focused_ e _Disabled_.
```
Faltou incluir o estado _Read-only,_ 

_Normal_, _Hover_, _Focused_, _Disabled_ e _Read-only_

edioli commented on packages/textarea/src/stories/code.mdx:
```diff
@@ -0,0 +1,49 @@
+
+> Para criar um `textarea group`, é importante adicionar os [controles via teclado indicados pela W3C
+> para o textarea group ](https://www.w3.org/TR/2017/WD-wai-aria-practices-1.1-20170628/examples/textarea/textarea-1/textarea-1.html) para manter a acessibilidade do componente.
+
```
Está correto essa parte de `textarea group`? O link está levando para uma pagina inexistente.

nicoleoliveira committed:
feat: correção da versão do typescript

Tem uma issue no typescript porque está travando o build.
https://github.com/microsoft/TypeScript/pull/45642
https://github.com/rollup/rollup/issues/4213

nicoleoliveira committed:
feat: correções na documentação apos validação

nicoleoliveira committed:
fix: correção no estilo disabled e safari

nicoleoliveira commented:
> * No estado **disabled**, por alguma razão não está pegando a cor do `--background-disabled`. Parece que a cor da variação do read-only está sobrepondo a do disabled.
> * Ficou faltando a opção de incluir um placeholder
> * No Safari, não está deixando redimensionar horizontalmente, como é original do componente (No Chrome está funcionando)
> 
> ![text-area-safari](https://user-images.githubusercontent.com/75502935/131923396-dd88ad8e-353b-4b1b-9b95-c3251bb0a56a.gif)

Correção concluída!

nicoleoliveira committed:
Update pr_ci.yml

nicoleoliveira committed:
fix: correção prettier

nicoleoliveira committed:
feat: remoção teste

nicoleoliveira committed:
feat: refactor

nicoleoliveira committed:
fix: prettier
